### PR TITLE
Remove dump and update show methods

### PR DESCRIPTION
**Pull request by musm**
suggestions welcome  ref https://github.com/JuliaIO/HDF5.jl/issues/366

### CODE REVIEW

**Comment by yuyichao** [comment]
I think this needs an doc update and possibly deprecation too (though I guess there shouldn't be an error with the old use of dump either)

**Comment by ggggggggg** [comment]
Looks pretty good, maybe a few tests?

File: src/HDF5.jl [hunk 1]
```diff
@@ -304,7 +304,22 @@ type HDF5File <: DataFile
...
 convert(::Type{Hid}, f::HDF5File) = f.id
-show(io::IO, fid::HDF5File) = isvalid(fid) ? print(io, "HDF5 data file: ", fid.filename) : print(io, "Closed HFD5 data file: ", fid.filename)
+function show(io::IO, fid::HDF5File)
+    if isvalid(fid)
+        if get(io, :compact, false)
+            print(io, "HDF5 data file: ", fid.filename)
+        else
+            print(io, "HDF5 data file: ", fid.filename, " (length ", length(fid), ")")
+            for k in names(fid)
+                print(io, "\n  ")
```
Comment (musm) [inline src/HDF5.jl]: perhaps it's not useful to print the names ?

**Comment by musm** [comment]
How do you deal with the deprecation in this case?  dump is currently imported and a method is added to the base function, which we want to avoid.

**Comment by musm** [comment]
Or should I just document the change and if so where ? Readme file?

**Comment by ggggggggg** [comment]
The more I think about it, the more I think printing them all out by default may not be very useful. For example, I often have 200+ groups in "/" in my hdf5 files, at that point I'd rather it print nothing. Maybe the functionality should just be renamed or removed instead of mapped onto part of `Base`? (and of course not error)

**Comment by musm** [comment]
good point I have removed the verbose printing

**Comment by musm** [comment]
What about renaming the old `dump` function to Base's `showall`?

**Comment by musm** [comment]
@ggggggggg do you have any opinions on the old `dump` function? 
It seems somewhat useful to get a quick glance at the structure of the hdf5 file, but I'm not sure if it's needed or if so what it should be called etc.

**Comment by ggggggggg** [comment]
I never used the previous `dump`, so I won't complain if it goes away. It does seem potentially useful, and I don't see any problem with keeping it.  The docs for `showall` say "Similar to show, except shows all elements of arrays." which doesn't seem to apply to anything other than array, which doesn't seem great for Base. `show_structure` seems better to me. It's a bit like a filesystem, but I don't think `walkdir` makes sense either.

**Comment by musm** [comment]
@ggggggggg `show_structure` sounds like a good name for this function

**Comment by musm** [comment]
Ok upon further thought I don't think the previous dump function is that useful nor robust. Deprecating it as in this PR seems the best strategy. 

I plan on merging this PR later this week sans objections.

**Pull request merged**